deploy: update CSI sidecars to latest versions available

iPraveenParihar opened the pull request:
<!-- Please take a look at our [Contributing](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#Code-contribution-workflow)
documentation before submitting a Pull Request!
Thank you for contributing to ceph-csi! -->

# Describe what this PR does #

Below sidecars are updated with latest available versions

csi-node-driver-registrar: v2.10.1
csi-resizer: v1.11.1
csi-provisioner: v5.0.1
csi-attacher: v4.6.1
csi-snapshotter: v8.0.1

Closes: #4632 

## Is there anything that requires special attention?
Feature gate Topology is now enabled by default and will be removed in a future release. [change log](https://github.com/kubernetes-csi/external-provisioner/blob/v5.0.1/CHANGELOG/CHANGELOG-5.0.md#no-really-you-must-read-this-before-you-upgrade).
Now, its CSI driver's responsibility to report `VOLUME_ACCESSIBILITY_CONSTRAINTS` if it support Topology. When the CSI driver reports it external provisioner will enables it, else disables it.

This PR has changes that remove `--feature-gate=Topology`  from RBD templates and RBD driver reports `VOLUME_ACCESSIBILITY_CONSTRAINTS`. Whereas, CephFS driver will not report `VOLUME_ACCESSIBILITY_CONSTRAINTS`  since Topology is not yet supported.


**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

iPraveenParihar committed:
deploy: update CSI sidecars to latest versions available

Below sidecars are updated with latest available versions

csi-node-driver-registrar: v2.10.1
csi-resizer: v1.11.1
csi-provisioner: v5.0.1
csi-attacher: v4.6.1
csi-snapshotter: v8.0.1

Signed-off-by: Praveen M <m.praveen@ibm.com>

Madhu-1 commented:
@iPraveenParihar rebase this PR?

iPraveenParihar commented:
> @iPraveenParihar rebase this PR?

yea, once this is merged https://github.com/ceph/ceph-csi/pull/4658.

iPraveenParihar commented:
@Mergifyio rebase

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.30

iPraveenParihar commented:
failed at 
```

Trying to pull registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.11.0...

Error: initializing source docker://registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.11.0: reading manifest v2.11.0 in registry.k8s.io/sig-storage/csi-node-driver-registrar: manifest unknown: Failed to fetch "v2.11.0"
```

https://github.com/kubernetes-csi/node-driver-registrar/issues/415

Madhu-1 commented:
> failed at
> 
> ```
> 
> Trying to pull registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.11.0...
> 
> Error: initializing source docker://registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.11.0: reading manifest v2.11.0 in registry.k8s.io/sig-storage/csi-node-driver-registrar: manifest unknown: Failed to fetch "v2.11.0"
> ```
> 
> [kubernetes-csi/node-driver-registrar#415](https://github.com/kubernetes-csi/node-driver-registrar/issues/415)

@iPraveenParihar lets update other except this one.

iPraveenParihar commented:
> @iPraveenParihar lets update other except this one.

using last working tag: [v2.10.1](https://github.com/kubernetes-csi/node-driver-registrar/releases/tag/v2.10.1)

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.30

iPraveenParihar commented:
@Mergifyio queue

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.27

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.28

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.29

iPraveenParihar commented:
/retest ci/centos/k8s-e2e-external-storage/1.30

iPraveenParihar commented:
/retest ci/centos/mini-e2e-helm/k8s-1.27

iPraveenParihar commented:
/retest ci/centos/mini-e2e-helm/k8s-1.28

iPraveenParihar commented:
/retest ci/centos/mini-e2e-helm/k8s-1.29

iPraveenParihar commented:
/retest ci/centos/mini-e2e-helm/k8s-1.30

iPraveenParihar commented:
/retest ci/centos/mini-e2e/k8s-1.30

nixpanic commented:
@mergifyio rebase

nixpanic commented:
@mergifyio queue

nixpanic commented:
**ci/centos/mini-e2e-helm/k8s-1.27** had a timeout while creating the 1st CephFS PV, please have a look at the logs and see if you can identify a reason for it ([logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/job/mini-e2e-helm_k8s-1.27/1009/display/redirect))

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.27

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.27

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.28

iPraveenParihar commented:
@Mergifyio refresh

iPraveenParihar commented:
@Mergifyio rebase

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.28

nixpanic commented:
@mergifyio rebase

nixpanic commented on charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml:
```diff
@@ -128,6 +128,7 @@ spec:
             - "--retry-interval-start=500ms"
             - "--extra-create-metadata=true"
             - "--feature-gates=HonorPVReclaimPolicy=true"
+            - "--feature-gates=Topology=false"
```
Why set it to `false` if it will be removed and enabled by default?

iPraveenParihar replied:
> Why set it to `false` if it will be removed and enabled by default?

CephFS doesn't support topology based provision.
For now, the feature gate default value is set to true (https://github.com/kubernetes-csi/external-provisioner/pull/1167) and moving forward the flag will be removed and its CSI Driver's responsibility to report the `VOLUME_ACCESSIBILITY_CONSTRAINTS` for topology based provisioning.

I think we can remove this now instead of `--feature-gates=Topology=false`?
https://github.com/ceph/ceph-csi/blob/e5927c03ed106563847d5c37cdbf9557a3e14a4e/internal/cephfs/identityserver.go#L54-L60

nixpanic replied:
yes, if I understand it correctly, I agree that it can be removed for CephFS. There should be no `--feature-gates=Topology=...` anywhere anymore.

iPraveenParihar replied:
I have updated the PR description regarding this change. PTAL

iPraveenParihar committed:
cephs: remove VOLUME_ACCESSIBILITY_CONSTRAINTS capabilities

This commit remove `VOLUME_ACCESSIBILITY_CONSTRAINTS` capabilities
from CephFS as topology based volume provisioning is not yet supported.

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar committed:
deploy: remove feature gate topology.

This commit removes the Topology feature gate as it is now enabled by default
and will be removed in a future release. It is CSI driver's responsibility to
report capability `VOLUME_ACCESSIBILITY_CONSTRAINTS` so that topology gets
enabled in external-provisioner. When driver doesn't report it,
external-provisioner disables topology support.

As of this change, Only RBD driver supports topology based volume provisioning
and it reports the `VOLUME_ACCESSIBILITY_CONSTRAINTS` capability,
enabling topology support in the external-provisioner.

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar committed:
helm: remove feature gate topology

This commit removes the Topology feature gate as it is now enabled
by default and will be removed in a future release.

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar committed:
e2e: remove enableTopologyInTemplate method

This commit removes `enableTopologyInTemplate` method as this is not
needed anymore.

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.30

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.30

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.30

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.30

nixpanic commented:
@mergifyio rebase

iPraveenParihar commented:
@Mergifyio rebase

iPraveenParihar commented:
@Mergifyio queue
